datahub-web-react/src/app/identity/group/CreateGroupModal.tsx

```diff
@@ -1,72 +1,90 @@
-import React, { useState } from 'react';
+import React, { useRef, useState } from 'react';
 import { message, Button, Input, Modal, Typography, Form, Collapse } from 'antd';
+import styled from 'styled-components';
 import { useCreateGroupMutation } from '../../../graphql/group.generated';
 import { useEnterKeyListener } from '../../shared/useEnterKeyListener';
 import { validateCustomUrnId } from '../../shared/textUtil';
 import analytics, { EventType } from '../../analytics';
 import { CorpGroup, EntityType } from '../../../types.generated';
+import { Editor as MarkdownEditor } from '../../entity/shared/tabs/Documentation/components/editor/Editor';
+import { ANTD_GRAY } from '../../entity/shared/constants';
 
 type Props = {
     onClose: () => void;
     onCreate: (group: CorpGroup) => void;
 };
 
+const StyledEditor = styled(MarkdownEditor)`
+    border: 1px solid ${ANTD_GRAY[4]};
+`;
+
 export default function CreateGroupModal({ onClose, onCreate }: Props) {
     const [stagedName, setStagedName] = useState('');
     const [stagedDescription, setStagedDescription] = useState('');
     const [stagedId, setStagedId] = useState<string | undefined>(undefined);
     const [createGroupMutation] = useCreateGroupMutation();
     const [createButtonEnabled, setCreateButtonEnabled] = useState(true);
     const [form] = Form.useForm();
+    const styledEditorRef = useRef<HTMLDivElement>(null);
 
     const onCreateGroup = () => {
-        createGroupMutation({
-            variables: {
-                input: {
-                    id: stagedId,
-                    name: stagedName,
-                    description: stagedDescription,
-                },
-            },
-        })
-            .then(({ data, errors }) => {
-                if (!errors) {
-                    analytics.event({
-                        type: EventType.CreateGroupEvent,
-                    });
-                    message.success({
-                        content: `Created group!`,
-                        duration: 3,
-                    });
-                    // TODO: Get a full corp group back from create endpoint.
-                    onCreate({
-                        urn: data?.createGroup || '',
-                        type: EntityType.CorpGroup,
+        const preventEnterModal =
```

Comment:
This is weird code - why do we have to do this?

If we have to do this, can we please add a comment explaining why?

Reply:
(feels hacky)

We also use this editor inside another component: DescriptionModal.tsx. Perhaps we can take an inspiration from there (I don't think we have to do that there)

![Screenshot 2023-12-13 at 4 53 57 PM](https://github.com/datahub-project/datahub/assets/17549204/5ac9ce77-3e23-4782-8f2e-281f886e083a)

Reply:
> This is weird code - why do we have to do this?
> 
> If we have to do this, can we please add a comment explaining why?

Actually Modal is close and saving data on pree senter.The likely scenario here is that the code is intended to prevent the default action (possibly saving data) when the "Enter" key is pressed, except when the focus is on the specified rich text editor (styledEditor). This might be necessary to allow users to create line breaks when press enter within the editor without triggering the default action.